packages/cli/src/public-api/v1/shared/middlewares/global.middleware.ts

```diff
@@ -74,6 +76,19 @@ export const validCursor = (
 	return next();
 };
 
+export const apiKeyHasScope = (
+	apiKeyScope: ApiKeyScope,
+	{ addGlobalScopeIfScopesDisabled } = { addGlobalScopeIfScopesDisabled: true },
+) => {
+	const emptyMiddleware = (_req: Request, _res: Response, next: NextFunction) => next();
+
+	return Container.get(License).isApiKeyScopesEnabled()
+		? Container.get(PublicApiKeyService).getApiKeyScopeMiddleware(apiKeyScope)
+		: addGlobalScopeIfScopesDisabled
+			? globalScope(apiKeyScope as Scope)
```

Comment:
Can we get rid of the casting here? I tried and found that `apiKeyScope` is not a subset of `Scope`, so sometimes we pass things in that don't exist.

While trying to figure out if we can get this to type check correctly or if this is hiding an issue I tried to understand what `addGlobalScopeIfScopesDisabled` is for, but couldn't quite figure it out. 
What requirement does it serve?